stdlib/builtin/hash.rbs

```diff
@@ -1,410 +1,1033 @@
+  #
+  # `key` should not have its value changed while it is in use as a key (an
+  # `unfrozen String` passed as a key will be duplicated and frozen).
+  #
+  #     a = "a"
+  #     b = "b".freeze
+  #     h = { a => 100, b => 200 }
+  #     h.key(100).equal? a #=> false
+  #     h.key(200).equal? b #=> true
+  #
+  alias store []=
 
   # Converts *hsh* to a nested array of `[` *key, value* `]` arrays.
-  # 
-  # ```ruby
-  # h = { "c" => 300, "a" => 100, "d" => 400, "c" => 300  }
-  # h.to_a   #=> [["c", 300], ["a", 100], ["d", 400]]
-  # ```
+  #
+  #     h = { "c" => 300, "a" => 100, "d" => 400, "c" => 300  }
+  #     h.to_a   #=> [["c", 300], ["a", 100], ["d", 400]]
+  #
   def to_a: () -> ::Array[[ K, V ]]
 
+  # Returns `self`. If called on a subclass of Hash, converts the receiver to a
+  # Hash object.
+  #
+  # If a block is given, the results of the block on each pair of the receiver
+  # will be used as pairs.
+  #
+  def to_h: () -> Hash[K, V]
+          | [A, B] () { (K, V) -> [ A, B ] } -> Hash[A, B]
+
   # Returns `self`.
+  #
   def to_hash: () -> ::Hash[K, V]
 
-  def to_proc: () -> ^(K) -> V
+  # Returns a Proc which maps keys to values.
+  #
+  #     h = {a:1, b:2}
+  #     hp = h.to_proc
+  #     hp.call(:a)          #=> 1
+  #     hp.call(:b)          #=> 2
+  #     hp.call(:c)          #=> nil
+  #     [:a, :b, :c].map(&h) #=> [1, 2, nil]
+  #
+  def to_proc: () -> ^(K) -> V?
+
+  alias to_s inspect
 
-  # Alias for: [inspect](Hash.downloaded.ruby_doc#method-i-inspect)
-  def to_s: () -> String
+  # Returns a new hash with the results of running the block once for every key.
+  # This method does not change the values.
+  #
+  #     h = { a: 1, b: 2, c: 3 }
+  #     h.transform_keys {|k| k.to_s }  #=> { "a" => 1, "b" => 2, "c" => 3 }
+  #     h.transform_keys(&:to_s)        #=> { "a" => 1, "b" => 2, "c" => 3 }
+  #     h.transform_keys.with_index {|k, i| "#{k}.#{i}" }
+  #                                     #=> { "a.0" => 1, "b.1" => 2, "c.2" => 3 }
+  #
+  # If no block is given, an enumerator is returned instead.
+  #
+  def transform_keys: () -> Enumerator[[ K, V ], Hash[untyped, V]]
+                    | [A] () { (K) -> A } -> Hash[A, V]
 
-  def value?: (V arg0) -> bool
+  # Invokes the given block once for each key in *hsh*, replacing it with the new
+  # key returned by the block, and then returns *hsh*. This method does not change
+  # the values.
+  #
+  #     h = { a: 1, b: 2, c: 3 }
+  #     h.transform_keys! {|k| k.to_s }  #=> { "a" => 1, "b" => 2, "c" => 3 }
+  #     h.transform_keys!(&:to_sym)      #=> { a: 1, b: 2, c: 3 }
+  #     h.transform_keys!.with_index {|k, i| "#{k}.#{i}" }
+  #                                      #=> { "a.0" => 1, "b.1" => 2, "c.2" => 3 }
+  #
+  # If no block is given, an enumerator is returned instead.
+  #
+  def transform_keys!: () -> Enumerator[[ K, V ], Hash[untyped, V]]
+                     | [A] () { (K) -> A } -> Hash[A, V]
```

Comment:
We give monomorphic types for `map!`. How about `() { (K) -> V } -> Hash[K, V]`?

Reply:
That makes sense. I updated `transform_keys!` and `transform_values!` typing.
b42db5ae93f30bec36cf6c43b0613017ef1a274a